### PR TITLE
PYTHON-3315 Remove index_key_id option from ClientEncryption::encrypt

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: bindings/python/test/test_mongocrypt.py [hunk 1]
```diff
@@ -517,9 +512,8 @@ def test_encrypt_indexed(self):
         key_path = 'keys/ABCDEFAB123498761234123456789012-local-document.json'
         index_key_path = 'keys/12345678123498761234123456789012-local-document.json'
```
Comment (blink1073) [inline bindings/python/test/test_mongocrypt.py:513]: Is `index_key_path` not needed at all anymore, here and in `test_rewrap_many_data_key`?

Comment (juliusgeo) [reply bindings/python/test/test_mongocrypt.py]: I believe you are correct. Removed.

**Review by blink1073 — APPROVED** [review]
LGTM!

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
test_rewrap_many_data_key is failing
```
 [2022/07/01 19:53:17.746] FAIL: test_rewrap_many_data_key (test.test_mongocrypt.TestExplicitEncryption)
 [2022/07/01 19:53:17.746] ----------------------------------------------------------------------
 [2022/07/01 19:53:17.746] Traceback (most recent call last):
 [2022/07/01 19:53:17.746]   File "z:\data\mci\6f5edcdc384b53e5e13b069cd762a784\libmongocrypt\bindings\python\test\test_mongocrypt.py", line 595, in test_rewrap_many_data_key
 [2022/07/01 19:53:17.746]     assert len(raw_doc['v']) == 2
 [2022/07/01 19:53:17.746] AssertionError
```

**Review by ShaneHarvey — APPROVED** [review]
LGTM!

**Pull request merged**